### PR TITLE
feat(blockly): added action + time trigger and condition json to xml …

### PR DESCRIPTION
**Pull request by Lukyio**
…conversion

### CODE REVIEW

File: client/types/automationType.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export interface AutomationPreview {
     id: string;
-    name: string;
+    alias: string;
```
Comment (Lukyio) [inline client/types/automationType.ts]: Should be alias right?

Comment (guyts96) [reply client/types/automationType.ts]: No, the swagger specifies that this field is called name. Please follow the documentation we have.

File: client/src/routes/automation/BlocklyComponent.svelte [hunk 2]
```diff
@@ -55,12 +54,49 @@
...
+            ],
+            action: [
+                {
+                    alias: '20:light.zipato_bulb_2_level:Lampe Z-wave Zipato',
+                    entity_id: 'light.zipato_bulb_2_level',
+                    service: 'light.turn_off',
+                },
+            ],
+            mode: 'single',
+            alias: 'test',
```
Comment (guyts96) [inline client/src/routes/automation/BlocklyComponent.svelte]:
```suggestion
            name: 'test',
```

File: client/src/services/blocklyService.ts [hunk 3]
```diff
@@ -120,45 +125,79 @@ export default class BlocklyService {
...
+                const attribute = xmlElement.getAttribute('entities_input');
+                const index = attribute != null ? parseInt(attribute, 10) : 0;
+                this.objectActionUpdateShape(index);
+            },
+
+            objectActionUpdateShape(index: number): void {
+                const newDropdown = entityWithServices[index]
                     ?.services.map((service: string) => [service.split('.')[1], service])
-                    ?? [['Action inconnue', 'Action inconnue']];
+                    ?? [['Action inconnu', 'Action inconnu']];
```
Comment (guyts96) [inline client/src/services/blocklyService.ts]: i18n + should be 'inconnue'

Comment (Lukyio) [reply client/src/services/blocklyService.ts]: Sorry merge conflict rest

File: client/src/routes/automation/esidomBlockGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            if (keys.includes('weekday') && !keys.includes('after') && !keys.includes('before')) {
+                const { weekday } = blocklyJSON;
+
+                xml += `
+                    <block type="time_condition_week">';
+                        ${weekday ? getWeekFields(weekday) : ''}
+                    </block>
+                `;
+                return xml;
+            } if (!keys.includes('weekday') && keys.includes('after') && keys.includes('before')) {
```
Comment (guyts96) [inline client/src/routes/automation/esidomBlockGenerator.ts:70]: else if ?

Comment (Lukyio) [reply client/src/routes/automation/esidomBlockGenerator.ts:70]: I did not do that : linter is the problem...

**Review by an-desco — APPROVED** [review]
💯

**Review by guyts96 — APPROVED** [review]

**Pull request merged**